internal/querycoordv2/task/merger.go

```diff
@@ -126,32 +104,44 @@ func (merger *Merger[K, R]) Add(task MergeableTask[K, R]) {
 	merger.waitQueue <- task
 }
 
-func (merger *Merger[K, R]) merge(id K, queue chan MergeableTask[K, R]) {
-	if merger.isStopped() {
-		return
-	}
-	if !merger.processors.Insert(id) {
-		return
+func (merger *Merger[K, R]) drain() {
+	for {
+		select {
+		case task := <-merger.waitQueue:
+			queue, ok := merger.queues[task.ID()]
+			if !ok {
+				queue = []MergeableTask[K, R]{}
+			}
+			queue = append(queue, task)
+			merger.queues[task.ID()] = queue
+		default:
+			return
+		}
 	}
-
-	merger.wg.Add(1)
-	go merger.mergeQueue(id, queue)
 }
 
-// mergeQueue merges tasks in the given queue,
-// it only processes tasks with the number of the length of queue at the time,
-// to avoid leaking goroutines
-func (merger *Merger[K, R]) mergeQueue(id K, queue chan MergeableTask[K, R]) {
-	defer merger.wg.Done()
-	defer merger.processors.Remove(id)
-
-	len := len(queue)
-	task := <-queue
-	for i := 1; i < len; i++ {
-		task.Merge(<-queue)
+func (merger *Merger[K, R]) triggerExecution(id K) {
+	tasks := merger.queues[id]
+	delete(merger.queues, id)
+
+	var task MergeableTask[K, R]
+	merged := 0
+	for i := 0; i < len(tasks); i++ {
+		if merged == 0 {
+			task = tasks[i]
+		} else {
+			task.Merge(tasks[i])
+		}
+		merged++
+		if merged >= int(Params.QueryCoordCfg.TaskMergeCap) {
+			merger.outCh <- task
+			merged = 0
+		}
+	}
+
+	if merged != 0 {
+		merger.outCh <- task
 	}
 
-	log.Info("merge tasks done",
-		zap.Any("mergeID", task.ID()))
-	merger.outCh <- task
+	log.Info("merge tasks done, trigger execution", zap.Any("mergeID", task.ID()))
```

Comment:
print task id is worthless, may we need to know what the target task comes from

Reply:
> 

taskID is not a number, but with several useful info such as colleciton and channel name

Reply:
I mean we should print this task merge from.